From 6ce31b7e2a2b8fbf764d6dd1efe8ba93ce225b5b Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Tue, 27 Feb 2024 08:08:02 +0200
Subject: [PATCH] [FSW-2066] fs1: update regs

Update Regs names and block base names due to specs_h15 changes

Change-Id: I7c9b96d7ddd1ccef605ee675e291bf2093668ab8
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/fs1/arc/asic.c b/arch/fs1/arc/asic.c
index f56ad98..ea23586 100644
--- a/arch/fs1/arc/asic.c
+++ b/arch/fs1/arc/asic.c
@@ -22,7 +22,7 @@
 {
 	u32 factor, psoc_div3_freq;
 
-	psoc_div3_freq = hl_pll_get_freq(mmIO_0_CFG_HALF_0_SOC_PLL_PCIE_CTRL_BASE, HL_PLL_DIV3);
+	psoc_div3_freq = hl_pll_get_freq(mmIO_0_CFG_HALF_0_SOC_0_PLL_0_PSOC_PLL_BASE, HL_PLL_DIV3);
 	factor = psoc_div3_freq / HL_SPI_FLASH_MAX_FREQ;
 
 	/* baud rate doesn't support odd number and must not be 0 */
@@ -39,7 +39,8 @@
 void hl_uart_init(void)
 {
 	/* As per Gaudi3 Frequencies.xlsx, PSOC PLL Div[3] drives uart */
-	u32 uart_clk_freq = hl_pll_get_freq(mmIO_0_CFG_HALF_0_SOC_PLL_PSOC_CTRL_BASE, HL_PLL_DIV3);
+	u32 uart_clk_freq = hl_pll_get_freq(mmIO_0_CFG_HALF_0_SOC_0_PLL_0_PSOC_PLL_BASE,
+										HL_PLL_DIV3);
 
 	regmap_writel(1, mmIO_0_CFG_HALF_0_PSOC_PSOC_CFG_PAD_CONF_BASE +
 			 mmPSOC_PAD_CONF_PAD_SEL_49);				/* UART0 RXD */
diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index 2a42418..eb1a0d7 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -130,7 +130,7 @@
 
 void device_init_ca53_cfg(void)
 {
-	ca53_cfg.arm_rst_control_reg = mmIO_0_CFG_HALF_0_SOC_CPU_CA53_CFG_BASE +
+	ca53_cfg.arm_rst_control_reg = mmIO_0_CFG_HALF_0_SOC_0_CPU_CA53_CFG_BASE +
 				       mmCA53_CFG_ARM_RST_CONTROL;
 }
 
@@ -205,7 +205,7 @@
 	regmap_writel(IO0_PSOC_ARC_MISC_ID, mmIO_0_CFG_HALF_0_PSOC_ARC_MISC_ARC_BASE +
 					    mmARC_ARC_NUM);
 	regmap_writel(IO0_PCIE_ARC_CXL_ID, mmIO_0_CFG_HALF_0_CXL_EP_PIF_ARC_CFG_BASE +
-					   mmPIF_ARC_ARC_NUM);
+					   mmARC_ARC_NUM);
 }
 
 __weak void hl_dev_psoc_pll_init(void) {}
diff --git a/arch/fs1/arc/pldm.c b/arch/fs1/arc/pldm.c
index 387540e..a55c55c 100644
--- a/arch/fs1/arc/pldm.c
+++ b/arch/fs1/arc/pldm.c
@@ -21,7 +21,7 @@
  */
 static struct hl_pll_cfg fs1_psoc_pll_cfg[] = {
 	{
-		.name = "PSOC", .base = mmIO_0_CFG_HALF_0_SOC_PLL_PSOC_CTRL_BASE,
+		.name = "PSOC", .base = mmIO_0_CFG_HALF_0_SOC_0_PLL_0_PSOC_PLL_BASE,
 		.nr = 1, .nf = 240, .od = 3, .nb = 2,
 		.div_factor = { 0, 1, 4, 3 },
 		.div_sel = { HL_PLL_PLL_CLK,
